go/vt/vtgate/engine/route.go

```diff
@@ -152,6 +153,32 @@ var routeName = map[RouteOpcode]string{
 	SelectDBA:         "SelectDBA",
 }
 
+var routeMetrics = map[RouteOpcode]*stats.Counter{
+	SelectUnsharded:   stats.NewCounter("RouteSelectUnsharded", "Counts of statements routed to an unsharded database."),
+	SelectEqualUnique: stats.NewCounter("RouteSelectEqualUnique", "Counts of statements routed to a single shard. Requires: A Unique Vindex, and a single Value."),
+	SelectEqual:       stats.NewCounter("RouteSelectEqual", "Counts of statements routed routed to one or more shards using a non-unique vindex. Requires: A Vindex, and a single Value."),
+	SelectIN:          stats.NewCounter("RouteSelectIN", "Counts of statements routed to one or more shards using a Vindex. Requires: A Vindex, and a Values list."),
+	SelectScatter:     stats.NewCounter("RouteSelectScatter", "Counts of statements routed with a scatter query to all shards of a keyspace"),
+	SelectNext:        stats.NewCounter("RouteSelectNext", "Counts of fetches from a sequence"),
+	SelectDBA:         stats.NewCounter("RouteSelectDBA", "Counts of a DBA statements"),
+}
+
+var (
+	scatterWidth = stats.NewCounter("RouteTotalScatterWidth", "Total width (shards routed to by a statement) of all statements routed")
+	queryCount   = stats.NewCounter("RouteQueryCount", "Total statements routed")
+)
+
+func init() {
+	stats.NewGaugeFunc("RoutesAverageScatterWidth", "Average width (shards routed to by a statement) of scatters routed", func() int64 {
```

Comment:
This should also not be needed because it's trivially calculated by monitoring tools.